RctGcs/ui/controls.py

```diff
@@ -262,16 +177,20 @@ def __advanced_settings(self):
         openSettings = ExpertSettingsDialog(self, self.option_vars)
         openSettings.exec_()
 
-    def validate_frequency(self, var: int):
+    def validate_frequency(self, value: int) -> bool:
         '''
         Helper function to ensure frequencies are within an appropriate
         range
         Args:
             var: An integer value that is the frequency to be validated
         '''
-        cntr_freq = self.__root._mav_model.getOption('SDR_center_freq')
-        samp_freq = self.__root._mav_model.getOption('SDR_sampling_freq')
-        if abs(var - cntr_freq) > samp_freq:
+        try:
+            model = MAVModel.get_model()
+        except NoActiveModel:
+            return
+        cntr_freq = model.getOption(Options.SDR_CENTER_FREQ)
+        samp_freq = model.getOption(Options.SDR_SAMPLING_FREQ)
+        if abs(value - cntr_freq) > samp_freq:
             return False
         return True
```

Comment:
unused method--should these checks be performed upon `add_entry` (in `widgets.py`)?